File: Lab4-middleware-hw/A2-subscription/Server/Services/IDeviceManager.cs
namespace SmartHome.Server
{
    public interface IDeviceManager
    {
        string ServerName { get; }
        List<DeviceBase> Devices { get; }
        DeviceBase GetDeviceById(string deviceId);
        DeviceBase GetDeviceByName(string deviceName);
        List<DeviceBase> GetDevicesbyType(DeviceType deviceType);
        bool TryControlDevice(ControlRequest request, out string message);
    }
}

File: Lab4-middleware-hw/A2-subscription/JavaServer/src/main/java/smarthome/server/IDeviceManager.java
package smarthome.server;

import java.util.List;
import smarthome.services.DeviceType;
import smarthome.services.ControlRequest;

public interface IDeviceManager {
    String getServerName();
    List<DeviceBase> getDevices();
    DeviceBase getDeviceById(String deviceId);
    DeviceBase getDeviceByName(String deviceName);
    List<DeviceBase> getDevicesByType(DeviceType deviceType);
    boolean tryControlDevice(ControlRequest request, OutParam<String> message);
}
